module/zfs/zfs_znode.c

```diff
@@ -1229,7 +1232,7 @@ zfs_rezget(znode_t *zp)
 
 	zp->z_mode = mode;
 
-	if (gen != zp->z_gen) {
+	if (gen != ZTOI(zp)->i_generation) {
```

Comment:
@tuxoko I was only suggesting the explicit cast here to remove any ambiguity about the comparison types.  Mainly for readability not for correctness.  The generation number shouldn't increment fast enough to exceed a 32-bit value in any reasonable length of time.